src/main/kotlin/org/opensearch/replication/ReplicationPlugin.kt

```diff
@@ -182,12 +191,14 @@ internal class ReplicationPlugin : Plugin(), ActionPlugin, PersistentTaskPlugin,
     }
 
     override fun getGuiceServiceClasses(): Collection<Class<out LifecycleComponent>> {
-        return listOf(Injectables::class.java,
+        return if (isPluginDisabled) Collections.emptyList()
+        else listOf(Injectables::class.java,
                 RemoteClusterRestoreLeaderService::class.java, RemoteClusterTranslogService::class.java)
     }
 
     override fun getActions(): List<ActionHandler<out ActionRequest, out ActionResponse>> {
-        return listOf(ActionHandler(GetChangesAction.INSTANCE, TransportGetChangesAction::class.java),
+        return if (isPluginDisabled) Collections.emptyList()
+        else listOf(ActionHandler(GetChangesAction.INSTANCE, TransportGetChangesAction::class.java),
```

Comment:
we might have to ensure the existing tasks fail gracefully after plugin is disabled

Reply:
Setting can't be modified dynamically and would take effect only on cluster restart. So, it'll just not spin up the task.

Reply:
I was referring to the following case:
- Plugin is enabled and replication is triggered on some of the indices.
- Now, the setting is updated to disable the plugin
- After cluster restart, the rest and transport actions will not be registered but the tasks might be still running.

We can discuss